sys/net/gnrc/routing/aodvv2/aodv.c

```diff
@@ -0,0 +1,543 @@
+ */
+void *fib_signal_handler_thread(void *arg)
+{
+    // TODO: warum wird der erst gestartet wenn ifconfig aufgerufen wird?
+    (void) arg;
+    msg_t fib_msgq[FIB_MSG_Q_SIZE];
+    ipv6_addr_t dest;
+    struct netaddr na_dest;
+
+    msg_init_queue(fib_msgq, FIB_MSG_Q_SIZE);
+    int err = fib_register_rp(&gnrc_ipv6_fib_table, &aodvv2_prefix.u8[0], aodvv2_address_type_size);
+    if ( err != 0) {
+        DEBUG("ERROR: cannot register at fib, error code:\n");
+        exit(1);
+    }
+
+    while (true) {
+        msg_t fib_msg;
+        msg_receive(&fib_msg);
+
+        if (fib_msg.type == FIB_MSG_RP_SIGNAL_UNREACHABLE_DESTINATION) {
+            rp_address_msg_t* rp_msg = (rp_address_msg_t*)fib_msg.content.ptr;
+            if (rp_msg->address_size == sizeof(ipv6_addr_t)) {
+                /* We currently only support IPv6*/
+                memcpy(&dest, rp_msg->address, rp_msg->address_size);
+                /* Reply to the FIB so that it can stop blocking */
+                msg_reply(&fib_msg, &fib_msg);
+
+                /* perform/initiate a rreq for dst here*/
+                ipv6_addr_t_to_netaddr(&dest, &na_dest);
+
+                aodvv2_seqnum_t seqnum = seqnum_get();
+                seqnum_inc();
+
+                /* Build new RREQ */
+                struct aodvv2_packet_data rreq_data = (struct aodvv2_packet_data) {
+                    .hoplimit = AODVV2_MAX_HOPCOUNT,
+                    .metricType = _metric_type,
+                    .origNode = (struct node_data) {
+                        .addr = na_local,
+                        .metric = 0,
+                        .seqnum = seqnum,
+                    },
+                    .targNode = (struct node_data) {
+                        .addr = na_dest,
+                    },
+                    .timestamp = (timex_t) {0,0} /* this timestamp is never used, it exists
+                                                  * merely to make the compiler shut up */
+                };
+
+                AODV_DEBUG("\tstarting route discovery towards %s... \n",
+                      ipv6_addr_to_str(addr_str, &dest, IPV6_ADDR_MAX_STR_LEN));
+                aodv_send_rreq(&rreq_data);
+            }
+            else {
+                /* Reply to the FIB so that it can stop blocking */
+                msg_reply(&fib_msg, &fib_msg);
+            }
+        }
+    }
+    return NULL;
+}
+
+void aodv_send_rreq(struct aodvv2_packet_data *packet_data)
+{
+    /* Make sure only one thread is dispatching a RREQ at a time */
+    mutex_lock(&rreq_mutex);
+
+    struct aodvv2_packet_data *pd = malloc(sizeof(struct aodvv2_packet_data));
+    memcpy(pd, packet_data, sizeof(struct aodvv2_packet_data));
+
+    struct rreq_rrep_data *rd = malloc(sizeof(struct rreq_rrep_data));
+    *rd = (struct rreq_rrep_data) {
+        .next_hop = &na_mcast,
+        .packet_data = pd,
+    };
+
+    struct msg_container *mc = malloc(sizeof(struct msg_container));
```

Comment:
If I see it right, the `msg_container` is the easiest to be stripped, since it mimics the structure of our `msg_t`.
We could probably exchange it by:
``` C
msg_t msg;
msg.type = RFC5444_MSGTYPE_RREQ;
msg.content.ptr = rd;
```